hw/hal/include/hal/hal_os_tick.h

```diff
@@ -49,6 +49,12 @@ void os_tick_init(uint32_t os_ticks_per_sec, int prio);
  */
 void os_tick_idle(os_time_t n);
 
+/**
+ * Calculate the OS tick parameters.
+ *
+ * @param os_ticks_per_sec The number of OS ticks per second
+ */
+void hal_os_tick_calc_params(uint32_t os_ticks_per_sec);
```

Comment:
Looking at the other services in hal_os_tick.c, I still believe this function belongs here. This is a new addition to the API, for the OS tick dividers to be re-tuned (case in point, after an oscillator re-calibration). It can be implemented for the other MCU's as well, on a need-to-do basis.

Reply:
I'm not convinced.
There is no code (in public repository) that requires this prototype. Even file in this commit does not need this.
If I was to implement this function for other MCU I would not know what is the purpose, why OS needs it and when it should be called.
To make your case it would be nice to have other code that utilizes this functionality. For now this function could be static in your hal_os_tick.c without any prototype.

Reply:
As a matter of fact, the reason I went thru the trouble of carving this out into a function & adding it to the API is that I wish to use it from our code.

Reply:
I don't really see why external code would need to call this. This calculates parameters based on current lpclk frequency so it just has to be done after lpclk frequency has changed e.g. after calibration so it can be called from that routine directly.

Reply:
Actually calling this from  `da1469x_lpclk_updated` should be enough.

Reply:
Agreed & done.